sklearn/utils/deprecation.py

```diff
@@ -62,9 +62,9 @@ def _decorate_class(self, cls):
 
         new = cls.__new__
 
-        def wrapped(*args, **kwargs):
+        def wrapped(cls, *args, **kwargs):
             warnings.warn(msg, category=FutureWarning)
-            return new(*args, **kwargs)
+            return new(cls)
```

Comment:
If `*args` are not passed up, then the following fails:

```python
@deprecated("a message")
class MockClass6:
    def __new__(cls, *args, **kwargs):
        assert args[0] == 42
        # if super() is object, then only `cls` can be passed up
        return super().__new__(cls)

    def __init__(self, a):
        self.a = a

def test_deprecated():
    with pytest.warns(FutureWarning, match="a message"):
        MockClass6(42)
``` 

I do not see a great way around it besides catching the error and trying again:

```python
def wrapped(cls, *args, **kwargs):
    warnings.warn(msg, category=FutureWarning)
    try:
        return new(cls, *args, **kwargs)
    except TypeError:
        # object accepts a single argument
        return new(cls)
```

Reply:
This makes me wonder if we really want to move this logic to `__new__`. It's adding complexity somewhere else, only hoping weird things don't happen to `__new__` in the child class I guess.

Since this is about deprecations and changes in parameters, should we just mark the original issue as "won't fix" and expect third party developers to handle it themselves?